paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -113,154 +112,193 @@ class PaparazziPlugin : Plugin<Project> {
+      PrepareResourcesTask::class.java
+    ) { task ->
+      val android = project.extensions.getByType(BaseExtension::class.java)
+      val nonTransitiveRClassEnabled =
+        (project.findProperty("android.nonTransitiveRClass") as? String)?.toBoolean() ?: true
+
+      task.packageName.set(android.packageName())
+      task.artifactFiles.from(packageAwareArtifactFiles)
+      task.nonTransitiveRClassEnabled.set(nonTransitiveRClassEnabled)
+      task.targetSdkVersion.set(android.targetSdkVersion())
+      task.compileSdkVersion.set(android.compileSdkVersion())
+      task.mergeAssetsOutputDir.set(buildDirectory.asRelativePathString(mergeAssetsProvider))
+      task.mergeResourcesOutputDir.set(buildDirectory.asRelativePathString(mergeResourcesProvider))
+      localResourceDirs?.let {
+        task.projectResourceDirs.from(it)
       }
+      task.moduleResourceDirs.from(moduleResourceDirs)
+      task.aarExplodedDirs.from(aarExplodedDirs)
+      localAssetDirs?.let {
+        task.projectAssetDirs.from(it)
+      }
+      task.projectAssetDirs.from(moduleAssetDirs)
+      task.aarAssetDirs.from(aarAssetDirs)
+      task.paparazziResources.set(buildDirectory.file("intermediates/paparazzi/${variant.name}/resources.txt"))
+    }
+
+    val testVariantSlug = testVariant.name.capitalize(Locale.US)
 
-      project.plugins.withType(KotlinMultiplatformPluginWrapper::class.java) {
-        val multiplatformExtension =
-          project.extensions.getByType(KotlinMultiplatformExtension::class.java)
-        check(multiplatformExtension.targets.any { target -> target is KotlinAndroidTarget }) {
-          "There must be an Android target configured when using Paparazzi with the Kotlin Multiplatform Plugin"
+    // TODO this task isn't actually produced by the JavaBasePlugin, this is a Kotlin task
+    project.plugins.withType(JavaBasePlugin::class.java) {
+      project.tasks
+        .matching {
+          it.name == "compile${testVariantSlug}JavaWithJavac"
         }
-        project.tasks.named("compile${testVariantSlug}KotlinAndroid")
-          .configure { it.dependsOn(writeResourcesTask) }
-      }
+        .configureEach { it.dependsOn(writeResourcesTask) }
+    }
 
-      project.plugins.withType(KotlinAndroidPluginWrapper::class.java) {
-        project.tasks.named("compile${testVariantSlug}Kotlin")
-          .configure { it.dependsOn(writeResourcesTask) }
+    project.plugins.withType(KotlinMultiplatformPluginWrapper::class.java) {
+      val multiplatformExtension =
+        project.extensions.getByType(KotlinMultiplatformExtension::class.java)
+      check(multiplatformExtension.targets.any { target -> target is KotlinAndroidTarget }) {
+        "There must be an Android target configured when using Paparazzi with the Kotlin Multiplatform Plugin"
       }
+      project.tasks
+        .matching { it.name == "compile${testVariantSlug}KotlinAndroid" }
+        .configureEach { it.dependsOn(writeResourcesTask) }
```

Comment:
yep it's a lazier approach and a little more correct, as afterEvaluate doesn't necessarily guarantee it since there can be multiple afterEvaluate levels